### DISCUSSION TITLE
Problem installing to the OneDrive Client Version v2.5.0

### DISCUSSION THREAD

**Discussion by Fertmeneses** [abraunegg/onedrive#2449] [Q&A] [answered]
Hello, I had the OneDrive Client Version v2.3 before and I'm trying to install **v2.5.0.** I'm having problems with the installation...

My OS is **Ubuntu 22.04.2 LTS** and I followed the instructions in https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md#distribution-ubuntu-2204 to install onedrive.

This is what I've done so far:

1. I followed the instructions from https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md.

2. I run `source ~/dlang/dmd-2.104.1/activate` without any problems.

3. I change directory to /onedrive and run `./configure,` with the following output:

```
(dmd-2.104.1)(base) meneses@6400l-212660-l:~/onedrive$ ./configure
checking for a BSD-compatible install... /usr/bin/install -c
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for dmd... dmd
checking version of D compiler... 2.104.1
checking for curl... yes
checking for sqlite... yes
configure: creating ./config.status
config.status: creating Makefile
config.status: creating contrib/pacman/PKGBUILD
config.status: creating contrib/spec/onedrive.spec
config.status: creating onedrive.1
config.status: creating contrib/systemd/onedrive.service
config.status: creating contrib/systemd/onedrive@.service
```

4. I run `make clean; make;` and I find a **problem** there. This is the output:
```
(dmd-2.104.1)(base) meneses@6400l-212660-l:~/onedrive$ make clean; make;
rm -f onedrive onedrive.o version
rm -rf autom4te.cache
rm -f config.log config.status
if [ -f .git/HEAD ] ; then \
	git describe --tags > version ; \
else \
	echo v2.4.25 > version ; \
fi
dmd -w -g -O -J. -L-lcurl -L-lsqlite3  -L-ldl src/config.d src/itemdb.d src/log.d src/main.d src/monitor.d src/onedrive.d src/qxor.d src/selective.d src/sqlite.d src/sync.d src/upload.d src/util.d src/progress.d src/arsd/cgi.d -ofonedrive
/home/unimelb.edu.au/meneses/miniconda3/bin/../lib/gcc/x86_64-conda-linux-gnu/12.1.0/../../../../x86_64-conda-linux-gnu/bin/ld: cannot find -lcurl: No such file or directory
collect2: error: ld returned 1 exit status
Error: linker exited with status 1
make: *** [Makefile:102: onedrive] Error 1
```
I'm stuck at this point and couldn't find any solution.

Could you please help me?

Thanks,

Fer

**Comment by abraunegg** [comment, marked as answer]
@Fertmeneses 
> My OS is Ubuntu 22.04.2 LTS and I followed the instructions in https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md#distribution-ubuntu-2204 to install onedrive.

For all Ubuntu platforms, whilst it is your choice to compile the application, it is *highly* recommended that you install the 'onedrive' application from the OpenSuSE Build Service Repository - as per [install.md](https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md):

![image](https://github.com/abraunegg/onedrive/assets/4956234/e8d821db-d1c1-440c-9b0a-d71048e17906)

Please read https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md to install on Ubuntu using the OpenSuSE Build Service packages.

This is also the *easiest* and *simplest* method to install the client, and will *keep* your client at the correct version, including when new releases are done.

Regarding this error:
```
cannot find -lcurl: No such file or directory
collect2: error: ld returned 1 exit status
Error: linker exited with status 1
```

This means that you have not installed the correct dependencies to build the application for Ubuntu.

> Hello, I had the OneDrive Client Version v2.3 before and I'm trying to install v2.5.0.

The v2.5.0 code has not been released yet - it is in heavy active development. Please refer to:
* https://github.com/abraunegg/onedrive/releases
* https://github.com/abraunegg/onedrive/discussions/2415

**Reply by Fertmeneses** [reply]
@abraunegg  
That worked perfectly!
Thanks a lot for the quick response and for this great repository.

**Comment by yezhengli-Mr9** [comment]
```java
source ~/dlang/dmd-2.106.0/activate
```
helps me to follow https://github.com/skilion/onedrive/issues/511. Thanks~

**Comment by abraunegg** [comment]
@yezhengli-Mr9 

To correctly install on Ubuntu you *must* follow this document: https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md
